### PR TITLE
Add Promoter icon into the WP Admin Bar

### PR DESCRIPTION
**Pull request by mitogh**
**Ticket** https://central.tri.be/issues/122515

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Each line in an array declaration must end in a comma | [Promoter_Connector.php:66](https://github.com/moderntribe/tribe-common/blob/b0e64066dfd07952430f67caa1d8c75f8f5a6c77/src/Tribe/Service_Providers/Promoter_Connector.php#L66)
:x: | Each line in an array declaration must end in a comma | [Promoter_Connector.php:68](https://github.com/moderntribe/tribe-common/blob/b0e64066dfd07952430f67caa1d8c75f8f5a6c77/src/Tribe/Service_Providers/Promoter_Connector.php#L68)
:x: | Missing space after array opener. | [Promoter_Connector.php:85](https://github.com/moderntribe/tribe-common/blob/b0e64066dfd07952430f67caa1d8c75f8f5a6c77/src/Tribe/Service_Providers/Promoter_Connector.php#L85)
:x: | There must be a single space after the first parenthesis | [Promoter_Connector.php:85](https://github.com/moderntribe/tribe-common/blob/b0e64066dfd07952430f67caa1d8c75f8f5a6c77/src/Tribe/Service_Providers/Promoter_Connector.php#L85)
:x: | There must be a single space after the first parenthesis | [Promoter_Connector.php:85](https://github.com/moderntribe/tribe-common/blob/b0e64066dfd07952430f67caa1d8c75f8f5a6c77/src/Tribe/Service_Providers/Promoter_Connector.php#L85)

> via jenkins/codesniffer-phpcs

**Comment by mitogh** [comment]
# Notes

Changes took place at the top of the WP Admin Bar.

## Before

screen shot 2019-02-14 at 13 15 12


## After

screen shot 2019-02-14 at 13 15 40

File: src/Tribe/Service_Providers/Promoter_Connector.php [hunk 1]
```diff
@@ -33,5 +33,71 @@ private function hook() {
...
+
+	/**
+	 * Add Admin Bar link to the promoter website
+	 *
+	 * @param $wp_admin_bar
+	 */
+	public function add_promoter_logo_on_admin_bar( $wp_admin_bar ) {
+		/** @var Tribe__Promoter__PUE $pue */
+		$pue = tribe( 'promoter.pue' );
+		if ( ! $pue->has_valid_license() ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Service_Providers/Promoter_Connector.php]: This method isn't cached, I have concerns about hitting this every admin page load

Comment (mitogh) [reply src/Tribe/Service_Providers/Promoter_Connector.php]: Oh! Right, I think we can replace it with `has_license_key` instead.

Let me know what do you think?

Comment (mitogh) [reply src/Tribe/Service_Providers/Promoter_Connector.php]: cc @sc0ttkclark

Comment (sc0ttkclark) [reply src/Tribe/Service_Providers/Promoter_Connector.php]: Yes that sounds much better

Comment (sc0ttkclark) [inline src/Tribe/Service_Providers/Promoter_Connector.php]:
```suggestion
		if ( ! $pue->has_license_key() ) {
```

File: src/Tribe/Service_Providers/Promoter_Connector.php [hunk 2]
```diff
@@ -33,5 +33,71 @@ private function hook() {
...
 		tribe( 'promoter.pue' );
+
+		// The usage of a high priority so we can push the icon to the end
+		add_action( 'admin_bar_menu', array( $this, 'add_promoter_logo_on_admin_bar' ), 1000 );
+		add_action( 'tribe_common_loaded', array( $this, 'add_promoter_assets' ) );
+	}
+
+	/**
+	 * Add Admin Bar link to the promoter website
+	 *
```
Comment (sc0ttkclark) [inline src/Tribe/Service_Providers/Promoter_Connector.php]:
```suggestion
	 *
	 * @since TBD
	 *
```

**Comment by mitogh** [comment]
This one is ready for another review thanks @sc0ttkclark .

**Pull request merged**